src/sql/workbench/parts/notebook/notebook.component.ts

```diff
@@ -582,4 +582,51 @@ export class NotebookComponent extends AngularDisposable implements OnInit, OnDe
 		}
 	}
 
+	getSections(): INotebookSection[] {
+		return this.getSectionElements();
+	}
+
+	private getSectionElements(): NotebookSection[] {
+		let headers: NotebookSection[] = [];
+		let el: HTMLElement = this.container.nativeElement;
+		let headerElements = el.querySelectorAll('h1, h2, h3, h4, h5, h6');
+		for (let i = 0; i < headerElements.length; i++) {
+			let headerEl = headerElements[i] as HTMLElement;
+			if (headerEl['id']) {
```

Comment:
can you explain this line a bit?

Reply:
So, the ID is the "slug" (finding new names from Markdown extensions TableOfContentsProvider) that is auto-generated for this header. Without this we can't do relative links since that's how the `#` link works in HTML. I expect this to be there, else someone output random H1 in an output or something and we should ignore.